File: frontend/src/types/copilot.ts
export interface CopilotSourceCitation {
  source_type: "FIRESTORE_OPERATIONAL" | "BIGQUERY_HISTORICAL" | "VERTEX_FORECAST"
  entity_id: string
  label: string
  value_referenced: string
}

export interface CopilotActionLink {
  label: string
  path: string
  icon_name: string
}

export interface CopilotQueryRequest {
  query: string
  facility_id?: string
  role?: string
}

export interface CopilotQueryResponse {
  query: string
  answer: string
  confidence_score: number
  citations: CopilotSourceCitation[]
  suggested_followups: string[]
  action_links: CopilotActionLink[]
  answered_at: string
  guardrails_passed: boolean
}

export interface ChatMessage {
  id: string
  sender: "user" | "copilot"
  text: string
  timestamp: string
  responsePayload?: CopilotQueryResponse
  isTyping?: boolean
}

File: backend/app/schemas/copilot.py
from typing import List, Optional, Dict, Any
from pydantic import BaseModel, Field

class CopilotQueryRequest(BaseModel):
    query: str = Field(..., min_length=2)
    facility_id: Optional[str] = None
    role: Optional[str] = None

class CopilotSourceCitation(BaseModel):
    source_type: str  # "FIRESTORE_OPERATIONAL", "BIGQUERY_HISTORICAL", "VERTEX_FORECAST"
    entity_id: str
    label: str
    value_referenced: str

class CopilotActionLink(BaseModel):
    label: str
    path: str
    icon_name: str

class CopilotQueryResponse(BaseModel):
    query: str
    answer: str
    confidence_score: float = Field(..., ge=0.0, le=1.0)
    citations: List[CopilotSourceCitation] = []
    suggested_followups: List[str] = []
    action_links: List[CopilotActionLink] = []
    answered_at: str
    guardrails_passed: bool = True
